### PR TITLE
Set $HOME to REALHOME and ensure xdg-user-dirs are created and used.

### PR DESCRIPTION
**Pull request by kenvandine**
This improves the user experience in nautilus and xdg-desktop-portals

### CODE REVIEW

File: scripts/desktop-launch.sh [hunk 1]
```diff
@@ -1,8 +1,12 @@
-#!/bin/sh
+#!/bin/bash
 
-exec gdbus call \
-     --session \
-     --dest io.snapcraft.Launcher \
-     --object-path /io/snapcraft/PrivilegedDesktopLauncher \
-     --method io.snapcraft.PrivilegedDesktopLauncher.OpenDesktopEntry \
-     "$1"
```
Comment (sergio-costas) [inline scripts/desktop-launch.sh:8]: Mmm... This script is already being used from `scripts/make-launchers.sh`. Removing this maybe could make it fail...

File: scripts/desktop-launch.sh [hunk 2]
```diff
@@ -1,8 +1,12 @@
-#!/bin/sh
+#!/bin/bash
```
Comment (sergio-costas) [inline scripts/desktop-launch.sh:1]: Is this change required? I tested the new lines with /bin/sh (which points to dash) and they work fine.

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -121,7 +121,21 @@ plugs:
       - /etc/pulse
       - /etc/shells
       - /etc/xdg/autostart
+      - /etc/xdg/user-dirs.conf
+      - /etc/xdg/user-dirs.defaults
```
Comment (jhenstridge) [inline snap/snapcraft.yaml:125]: We'll need to get the snap declaration updated to get this through. It might be lower effort to add this access to as permanent slot rules for the desktop interface.

**Review by jhenstridge — CHANGES_REQUESTED** [review]
I think this is a reasonable change to make. I've added a few suggestions to simplify things, and how we could avoid the need for getting the snap declaration updated.

File: snap/snapcraft.yaml (hunk 3 above)
Comment (kenvandine) [reply snap/snapcraft.yaml:131]: > We'll need to get the snap declaration updated to get this through. It might be lower effort to add this access to as permanent slot rules for the desktop interface.

@jhenstridge would this be sufficient?  https://github.com/canonical/ubuntu-core-desktop-snapd/pull/12

**Review by jhenstridge — APPROVED** [review]
Looks good!
